python/ray/data/dataset.py

```diff
@@ -2418,46 +2418,66 @@ def write_parquet(
         ray_remote_args: Dict[str, Any] = None,
         **arrow_parquet_args,
     ) -> None:
-        """Write the dataset to parquet.
+        """Writes the :class:`~ray.data.Dataset` to parquet files under the provided ``path``.
 
-        This is only supported for datasets convertible to Arrow records.
-        To control the number of files, use :meth:`Dataset.repartition`.
+        The number of files is determined by the number of blocks in the dataset.
+        To control the number of number of blocks, call
+        :meth:`~ray.data.Dataset.repartition`.
 
-        Unless a custom block path provider is given, the format of the output
-        files will be {uuid}_{block_idx}.parquet, where ``uuid`` is an unique
-        id for the dataset.
-
-        Examples:
+        This method is only supported for datasets with records that are convertible to
+        pyarrow tables.
```

Comment:
```suggestion
        This method is only supported for datasets with rows that are convertible to
        pyarrow tables.
```

Reply:
Supernit: we don't convert rows to tables, right? We convert blocks (pandas DataFrames) to tables? Maybe something like this would be more precise:
```suggestion
        If pyarrow can't represent your data, this method errors.
```

Reply:
updated